### ISSUE TITLE
Bulk processing many files

### ISSUE THREAD

**Issue by priyadarshan** [fniessen/orgmk#4]
Hi,

We have a use-case where each org-file need to be split to many corresponding org-files, one per heading. Ultimately we end up having quite a few directories (one per original file), each containing a few dozens org files.

When there is a need to process many `org` files, is there a way for `orgmk` to load Emacs+org library only once for all the files, to speed things up?

It would be wonderful if `orgmk` could perhaps leverage some [async](https://github.com/jwiegley/emacs-async) library, in order to process a few org-files in parallel, since the bottleneck is the CPU, not the disk io. Right now Emacs uses only 1 core, even if there are a few available.

Thank you once again for `orgmk`!

**Comment by fniessen** [comment]
Would you like to provide a patch?

**Comment by priyadarshan** [comment]
I am just an absolute beginner with emacs lisp, I could not come close to use the async library, or other equivalent ways.

We have been using `orgmk` to process a few hundred files, with every editorial change, often a few times a day. It works very well, it is just that the org-mode conversion can be quite slow. Thanks!

**Comment by fniessen** [comment]
IIUC, the "loop" could/should be written in the shell script, not in ELisp.

You can always do it "externally" as well:

    for file in "$@"; do
        org2html $file;
    done

Does it address your question?

**Comment by fniessen** [comment]
BTW, if you just call "orgmk", it will just export the files (from the current directory) which have changed, so you don't need to list them in any way.

**Comment by priyadarshan** [comment]
That is useful, thank you for the clarification!

What I meant was the ability to run orgmk on a few files at the same time.

I realise that the bottleneck is not with orgmk, but with emacs itself.

No matter how many cores (on my machine I have 12), emacs will always use one.

That was why I thought calling orgmk on multiple files, using multiple emacs instances, would speed up the process considerably.

I tried to do it by hand, and I can see that each separate emacs instance consume a separate cpu core, so, I think it would work.

But I can't think about the details on how to achieve that from isncie emacs, or even from the shell.

**Comment by fniessen** [comment]
OK, I better understand your need now. Though, I'm not aware, neither, of ways to enable parallelism in the shell, so I'll close this for now.

Be free to come back to it if you have new information on this!

**Comment by priyadarshan** [comment]
Sorry for reviving this topic, but I have been able to fulfill my initial need by using [GNU parallel](https://www.gnu.org/software/parallel/) with great success, with an incantation similar to this:
```
find . -name '*.org' | parallel --progress '/Applications/Emacs.app/Contents/MacOS/Emacs -Q --batch --load=~/elisp/latex-export.el --visit={} --funcall z/headless-latex'
```
It cuts my org -> TeX processing time manyfold. See attached screenshot for a recent task I had to run several dozen time recently.

![screenshot 2016-05-29 at 09 56 17](https://cloud.githubusercontent.com/assets/1652143/15631994/be2126e6-2583-11e6-9e94-7f60e88d47c2.png)

Perhaps in the future there will be a way to add it to your wonderful `orgmk`.

Thank you once again for this extremely useful tool.
